pdg/src/builder.rs

```diff
@@ -57,7 +57,13 @@ impl EventKindExt for EventKind {
             Realloc { old_ptr, .. } => old_ptr,
             FromInt(lhs) => lhs,
             Alloc { ptr, .. } => ptr,
-            AddrOfLocal(lhs, _) => lhs,
+            AddrOfLocal(lhs, _, _) => lhs,
+            AddrOfConst(ptr, _) => ptr,
+            // Corner case: Offset(..) events with a base pointer of zero are special
+            // because the result might be an actual pointer, e.g., c2rust will
+            // emit a pointer increment `a += b` as `a = a.offset(b)` which we need
+            // to ignore here if `a == 0` which is equivalent to `a = b`.
+            Offset(0, _, ptr) => ptr,
```

Comment:
Could this be detected and handled earlier (such as by replacing the `Offset` with a copy operation) to avoid the special case here?  This sounds like the sort of thing that would require matching special cases in several other places, which could easily be forgotten.

Reply:
Moved it to the event handlers.